apps/dashboard/app/javascript/packs/files/globus.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Given a directory name, find the associated Globus endpoint
+ * @params {string} directory Directory name
+ * @return {string|undefined} Globus endpoint ID
+ */
+function getEndpoint(directory) {
+  for (const [prefix, endpoint] of Object.entries(globus_endpoints)) {
+    if(directory.startsWith(prefix)) {
+      return endpoint
+    }
+  }
+}
+
+/**
+ * Generate a link to the Globus transfer app
+ * @params {string} directory Directory name
+ * @return {string|null} Globus transfer app URL with directory and endpoint populated
+ */
+export function getGlobusLink(directory) {
+  let endpoint = getEndpoint(directory);
+  let url = null;
+
+  if (endpoint) {
+    url = "https://app.globus.org/file-manager?origin_id=" + endpoint + "&origin_path=" + directory;
```

Comment:
There seems to be some issue with the root of the collection differing from the root of the actual filesystem.

This is what's being generated for my HOME directory that's shared in globus.

https://app.globus.org/file-manager?origin_id=06758ab7-9cb0-4f90-b79f-098b41e90dcd&origin_path=/users/PZS0714/johrstrom

But this is is what it should actually be:

https://app.globus.org/file-manager?origin_id=06758ab7-9cb0-4f90-b79f-098b41e90dcd&origin_path=%2F

Just the `/` root. This link above for me is the root of my own HOME directory. So it's `/` in Globus but it's `/users/PZS0714/johrstrom` on the actual file system.

Reply:
Thank you for taking a look at this. I tested your recommended html/css changes and those worked well. That's also a good point about the filesystem and globus endpoint paths potentially differing. I'll make some changes to account for that.